qiskit/pulse/schedule.py

```diff
@@ -167,74 +172,144 @@ def _instructions(self, time: int = 0):
                 :class:`~qiskit.pulse.Instruction`
                 starts at and the flattened :class:`~qiskit.pulse.Instruction` s.
         """
-        for insert_time, child_sched in self._children:
+        for insert_time, child_sched in self.children:
             yield from child_sched._instructions(time + insert_time)
 
-    def union(self, *schedules: Union[ScheduleComponent, Tuple[int, ScheduleComponent]],
-              name: Optional[str] = None) -> 'Schedule':
-        """Return a new schedule which is the union of both ``self`` and ``schedules``.
+    def union(self,
+              *schedules: Union[ScheduleComponent, Tuple[int, ScheduleComponent]],
+              name: Optional[str] = None,
+              mutate: bool = False
+              ) -> 'Schedule':
+        """Return a schedule which is the union of both ``self`` and ``schedules``.
 
         Args:
-            *schedules: Schedules to be take the union with this ``Schedule``.
+            schedules: Schedules to be take the union with this ``Schedule``.
             name: Name of the new schedule. Defaults to the name of self.
+            mutate: Perform operation by mutating this schedule. Otherwise return
+                a new ``Schedule``.
         """
         warnings.warn("The union method is deprecated. Use insert with start_time=0.",
                       DeprecationWarning)
-        if name is None:
-            name = self.name
-        new_sched = Schedule(name=name)
-        new_sched._insert(0, self)
-        for sched_pair in schedules:
-            if not isinstance(sched_pair, tuple):
-                sched_pair = (0, sched_pair)
-            new_sched._insert(sched_pair[0], sched_pair[1])
-        return new_sched
+        return self.insert(0, *schedules, name=name, mutate=mutate)
 
-    def _insert(self, start_time: int, schedule: ScheduleComponent) -> 'Schedule':
-        """Mutably insert `schedule` into `self` at `start_time`.
+    # pylint: disable=arguments-differ
```

Comment:
Doesn't this imply there's an issue with `ScheduleComponent`?

Reply:
There's a lot of debate on the internet about whether this particular linting flag is useful. What I've done doesn't even violate LSP (we've added a new default argument not removed any inherited arguments). `mutate` certainly doesn't make sense for an `Instruction`.